internal/pkg/ecs/run_task_request.go

```diff
@@ -0,0 +1,216 @@
+	cluster       string
+
+	containerInfo
+}
+
+type containerInfo struct {
+	image      string
+	entryPoint []string
+	command    []string
+	envVars    map[string]string
+	secrets    map[string]string
+}
+
+// RunTaskRequestFromECSService populates a RunTaskRequest with information from an ECS service.
+func RunTaskRequestFromECSService(client ECSServiceDescriber, cluster, service string) (*RunTaskRequest, error) {
+	networkConfig, err := client.NetworkConfiguration(cluster, service)
+	if err != nil {
+		return nil, fmt.Errorf("retrieve network configuration for service %s in cluster %s: %w", service, cluster, err)
+	}
+
+	svc, err := client.Service(cluster, service)
+	if err != nil {
+		return nil, fmt.Errorf("retrieve service %s in cluster %s: %w", service, cluster, err)
+	}
+
+	taskDefNameOrARN := aws.StringValue(svc.TaskDefinition)
+	taskDef, err := client.TaskDefinition(taskDefNameOrARN)
+	if err != nil {
+		return nil, fmt.Errorf("retrieve task definition %s: %w", taskDefNameOrARN, err)
+	}
+
+	if len(taskDef.ContainerDefinitions) > 1 {
+		return nil, fmt.Errorf("found more than one container in task definition: %s", taskDefNameOrARN)
+	}
+
+	containerName := aws.StringValue(taskDef.ContainerDefinitions[0].Name)
+	containerInfo, err := containerInformation(taskDef, containerName)
+	if err != nil {
+		return nil, err
+	}
+
+	return &RunTaskRequest{
+		networkConfiguration: *networkConfig,
+		executionRole:        aws.StringValue(taskDef.ExecutionRoleArn),
+		taskRole:             aws.StringValue(taskDef.TaskRoleArn),
+		containerInfo:        *containerInfo,
+		cluster:              cluster,
+	}, nil
+}
+
+// RunTaskRequestFromECSService populates a RunTaskRequest with information from a Copilot service.
+func RunTaskRequestFromService(client ServiceDescriber, app, env, svc string) (*RunTaskRequest, error) {
+	networkConfig, err := client.NetworkConfiguration(app, env, svc)
+	if err != nil {
+		return nil, fmt.Errorf("retrieve network configuration for service %s: %w", svc, err)
+	}
+
+	cluster, err := client.ClusterARN(app, env)
+	if err != nil {
+		return nil, fmt.Errorf("retrieve cluster ARN created for environment %s in application %s: %w", env, app, err)
+	}
+
+	taskDef, err := client.TaskDefinition(app, env, svc)
+	if err != nil {
+		return nil, fmt.Errorf("retrieve task definition for service %s: %w", svc, err)
+	}
+
+	containerName := svc // NOTE: refer to workload's CloudFormation template. The container name is set to be the workload's name.
+	containerInfo, err := containerInformation(taskDef, containerName)
+	if err != nil {
+		return nil, err
+	}
+
+	return &RunTaskRequest{
+		networkConfiguration: *networkConfig,
+		executionRole:        aws.StringValue(taskDef.ExecutionRoleArn),
+		taskRole:             aws.StringValue(taskDef.TaskRoleArn),
+		containerInfo:        *containerInfo,
+		cluster:              cluster,
+	}, nil
+}
+
+// String stringifies a RunTaskRequest.
+func (r RunTaskRequest) String() string {
```

Comment:
I don't think we should implement the [fmt.Stringer](https://golang.org/pkg/fmt/#Stringer) interface if it can't be used across packages. For example, if we do `fmt.Sprintf("%s", RunTaskRequest{...})` it's going to write `"copilot task run ..."` which is surprising.

Instead, we can create our own string method: `CLIString() string` and a `type CLIStringer interface` so that we can implement this interface across many data types

Reply:
addressed!